torchci/lib/clickhouse.ts

```diff
@@ -0,0 +1,49 @@
+import { createClient } from "@clickhouse/client";
+import { readFileSync } from "fs";
+
+export function getClickhouseClient() {
+  return createClient({
+    host: process.env.CLICKHOUSE_HUD_USER_URL ?? "http://localhost:8123",
+    username: process.env.CLICKHOUSE_HUD_USER_USERNAME ?? "default",
+    password: process.env.CLICKHOUSE_HUD_USER_PASSWORD ?? "",
+  });
+}
+
+export async function queryClickhouse(
+  query: string,
+  params: Record<string, unknown>
+) {
+  /**
+   * queryClickhouse
+   * @param query: string, the sql query
+   * @param params: Record<string, unknown>, the parameters to the query ex { sha: "abcd" }
+   */
+  const clickhouseClient = getClickhouseClient();
+  const res = await clickhouseClient.query({
+    query,
+    format: "JSONEachRow",
+    query_params: params,
+  });
+
+  return await res.json();
```

Comment:
I think you might need to manipulate the response here a bit to match the current format from Rockset (list of dictionary).  IIRC, ClickHouse returns columns and rows in two separate fields to reduce the payload side (and they can do that because the schema is fixed), i.e. https://github.com/pytorch/test-infra/pull/5393/files#diff-fbd9177049fad22046df3033a9fe62f8982861a79556524812b89feccfe11c8bR34.  I'm not sure if the client has already done that (probably not)

Reply:
I think it's fine actually, if you look at the console log in https://github.com/pytorch/test-infra/pull/5562 (doesn't actually work probably because of typing stuff), it looked like a list of dictionaries to me

Reply:
> I think you might need to manipulate the response here a bit to match the current format from Rockset (list of dictionary). IIRC, ClickHouse returns columns and rows in two separate fields to reduce the response size (and they can do that because the schema is fixed), i.e. https://github.com/pytorch/test-infra/pull/5393/files#diff-fbd9177049fad22046df3033a9fe62f8982861a79556524812b89feccfe11c8bR34. I'm not sure if the client has already done that (probably not)

CH [supports a variety of formats](https://clickhouse.com/docs/en/interfaces/formats). Pretty sure there is one that matches Rockset exactly.